SO-4642: Change module of inactivated content

molnarlaura opened the pull request.

Commit:
SO-4642: add test for changing module after inactivation

Commit:
SO-4642: change relationships' module after inactivation

Commit:
Revert "SO-4642: add test for changing module after inactivation"

This reverts commit ba83c999a4342adb9f98c6e02971dae36d8a1797.

Commit:
SO-4642: add test for changing module after inactivation

Commit:
SO-4642: remove extra line

Commit:
SO-4642: use CodeSystemURI

Commit:
SO-4642: change moduleId for all effected component

apeteri commented on snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/change/ComponentInactivationChangeProcessor.java:
```diff
@@ -93,13 +91,21 @@ public void process(StagingArea staging, RevisionSearcher searcher) throws IOExc
 				changedMembersByReferencedComponentId.put(((SnomedRefSetMemberIndexEntry) diff.newRevision).getReferencedComponentId(), diff);
 			});
 			
+			SnomedRefSetMemberIndexEntry memberForModule = staging.getChangedRevisions(SnomedRefSetMemberIndexEntry.class).findFirst().map(diff -> (SnomedRefSetMemberIndexEntry) diff.newRevision).get();
```
It seems like this member can come from anywhere. What is its relation to the description inactivation member we are going to use `moduleId` with?

molnarlaura replied:
As discussed, I leave the code as it is.

apeteri commented:
Unfortunately `snomed-e2e-tests` are failing on this branch (both here on GitHub and after merging in the latest changes from 7.x, locally). The primary issue is that some transaction contexts in the SNOMED CT store do not provide an instance of `ModuleIdProvider`, like the RF2 importer and code system creation commits, including the creation of the primary code system in the terminology repository initializer. 

Tests can not start without this resolved.

apeteri commented on snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/components/SnomedConceptApiTest.java:
```diff
@@ -945,4 +927,51 @@ public void testPendingMoveUpdateOnConceptWithSetDescriptions() {
+		final String conceptId = createNewConcept(branchPath);
+		String sourceRelationshipId = createNewRelationship(branchPath, conceptId, Concepts.HAS_DOSE_FORM, Concepts.MODULE_SCT_MODEL_COMPONENT);
+		String destinationRelationshipId = createNewRelationship(branchPath, Concepts.MODULE_SCT_MODEL_COMPONENT, Concepts.HAS_DOSE_FORM, conceptId);
+		CodeSystemURI codeSystemURI = CodeSystemURI.branch(SnomedTerminologyComponentConstants.SNOMED_SHORT_NAME, branchPath.getPath());
```
`branchPath` is absolute, so this test looks for a branch path named `MAIN/MAIN/...`, and so can not find what it was looking for.

molnarlaura replied:
Done

Commit:
SO-4642: remove unused content

Commit:
SO-4642: use always the new moduleId

Commit:
SO-4642: create new module for test

Commit:
SO-4642: initialize ModulIdProvider for requests

Commit:
Merge remote-tracking branch 'origin/7.x' into improvement/SO-4642-change-module-of-inactivated-content

Commit:
SO-4642: use ModuleIdProvider on every component

Commit:
SO-4642: register ModuleIdProvider in rather in getRequestConfigurer()

Commit:
SO-4642: use prepareCommit to use ModuleRequest's moduleIdProvider

Commit:
SO-4642: bind ModuleIdProvider to context

apeteri commented on snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/change/ComponentInactivationChangeProcessor.java:
```diff
@@ -93,23 +91,28 @@ public void process(StagingArea staging, RevisionSearcher searcher) throws IOExc
+			for (Hits<SnomedDescriptionIndexEntry> hits : searcher.scroll(Query.select(SnomedDescriptionIndexEntry.class)
 					.from(SnomedDescriptionIndexEntry.class)
-					.fields(SnomedDescriptionIndexEntry.Fields.ID, SnomedDescriptionIndexEntry.Fields.MODULE_ID)
+					.fields(SnomedDescriptionIndexEntry.Fields.ID)
```
`MODULE_ID` is still needed so the provider can work :)

cmark committed:
Merge remote-tracking branch 'origin/7.x' into improvement/SO-4642-change-module-of-inactivated-content

cmark committed:
SO-4642: fix remaining PR issues

Apply moduleId change during reactivation as well (at least for the
currently handled parts).

cmark reviewed (approved):
LGTM!
